staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -693,6 +724,29 @@ type PersistentVolumeClaimStatus struct {
 	// +mapType=granular
 	// +optional
 	AllocatedResourceStatuses map[ResourceName]ClaimResourceStatus `json:"allocatedResourceStatuses,omitempty" protobuf:"bytes,7,rep,name=allocatedResourceStatuses"`
+	// volumeAttributesClassName is the name of VolumeAttributesClass which is latest applied by
+	// the CSI driver. This field is not set by default but when modification is complete,
+	// it's set to the applied VolumeAttributesClass.
+	// This is an alpha field and requires enabling VolumeAttributesClass feature.
+	// +featureGate=VolumeAttributesClass
+	// +optional
+	VolumeAttributesClassName *string `json:"volumeAttributesClassName,omitempty" protobuf:"bytes,8,opt,name=volumeAttributesClassName"`
+	// VolumeAttributesModifyStatus stores status of modification operation. It can be in any of following states:
+	//  - PersistentVolumeClaimControllerModifyVolumeInProgress
+	//    State set when modify volume controller starts modifying the volume in control-plane.
+	//  - PersistentVolumeClaimControllerModifyVolumeFailed
+	//    State set when modify volume has failed in modify volume controller with a terminal error.
+	//    Transient errors such as timeout should not set this status and should leave VolumeAttributesModifyStatus
+	//    unmodified, so as modify volume controller can resume the volume modification.
+	// For example: if modifying a PVC's volume attributes, this field can be one of the following states:
+	//   - pvc.status.volumeAttributesModifyStatus = "ControllerModifyVolumeInProgress"
+	//   - pvc.status.volumeAttributesModifyStatus = "ControllerModifyVolumeFailed"
+	// When this field is not set, it means that no volume attributes modification operation is in progress for the given PVC.
+	//
+	// This is an alpha field and requires enabling VolumeAttributesClass feature.
+	// +featureGate=VolumeAttributesClass
+	// +optional
+	VolumeAttributesModifyStatus *VolumeAttributesClassStatus `json:"volumeAttributesModifyStatus,omitempty" protobuf:"bytes,9,opt,name=volumeAttributesModifyStatus,casttype=PersistentVolumeClaimVolumeAttributesModifyStatus"`
```

Comment:
If the controller identifies that the volumeAttributesClassName is set to a non-existant VolumeAttributesClass, or that the VolumeAttributesClass's DriveName is set to a non-existant driver name, what happens?

Reply:
If the referenced VolumeAttributeClass is not found, the status field will be set by `InProgress` and retry will be attempted in the next sync cycle. An warning/error event will be reported with the message `VolumeAttributesClass not found`. 

If the referenced driver name is not found, it means that the driver is not installed. In this case, the status field will be set by `InProgress` and retry will be attempted in the next sync cycle. An warning/error event will be reported.

/cc @sunnylovestiramisu 

What do you think?

Reply:
Yeah that was the intended behavior, but we are having another discussion: https://docs.google.com/document/d/1998hur3OZL1-bLyoqJdkzgDCOjqM8dZLJdzdnhDqMiU/edit?resourcekey=0-4miOrF1vQI9nc31xcb4I5w so that we do not retry indefinitely.